alpha/apps/kaltura/lib/kSimuliveUtils.php

```diff
@@ -302,4 +295,54 @@ public static function getClosestPlaybackTransitionTime($event, $time)
 		// we shouldn't arrive this if $time is inside event
 		return null;
 	}
+
+	/**
+	 * receiving array of entries, returning 3 asset arrays (flavors, captions, audio) 
+	 * s.t each array's i'th element is the asset array of the i'th entry assets.
+	 * @param array $sourceEntries
+	 * @return array
+	 */
+	public static function getSourceAssets($sourceEntries)
+	{
+		$entriesFlavorAssets = array();
+		$entriesCaptionAssets = array();
+		$entriesAudioAssets = array();
+		foreach ($sourceEntries as $srcEntry)
+		{
+			list($mainFlavorAssets, $mainCaptionAssets, $mainAudioAssets) = myEntryUtils::getEntryAssets($srcEntry);
+			$entriesFlavorAssets = array_merge($entriesFlavorAssets, array($mainFlavorAssets));
+			$entriesCaptionAssets = array_merge($entriesCaptionAssets, array($mainCaptionAssets));
+			$entriesAudioAssets = array_merge($entriesAudioAssets, array($mainAudioAssets));
+			$durations[] = self::roundDuration($srcEntry->getLengthInMsecs());
+		}
+		return array($entriesFlavorAssets, $entriesCaptionAssets, $entriesAudioAssets);
+	}
+
+	/**
+	 * receiving array of entries and event. returning array of durations s.t the i'th element is the duration of the 
+	 * i'th entry. if accumulated duration is exceeding the event's duration - the appropriate duration will be shorten
+	 * in accordance, and all the durations after will be shorten to '1'
+	 * @param array $sourceEntries
+	 * @param ILiveStreamScheduleEvent $event
+	 * @return array
+	 */
+	public static function getSourceDurations($sourceEntries, $event)
+	{
+		$durations = array();
+		$eventDuration = $event->getCalculatedEndTime() * self::SECOND_IN_MILLISECONDS - $event->getCalculatedStartTime() * self::SECOND_IN_MILLISECONDS;
+		$aggregatedDuration = 0;
+		foreach ($sourceEntries as $srcEntry)
+		{
+			$entryRoundedDuration = self::roundDuration($srcEntry->getLengthInMsecs());
+			$aggregatedDuration += $entryRoundedDuration;
+			if ($aggregatedDuration >= $eventDuration)
+			{
+				$durations[] = max($entryRoundedDuration - ($aggregatedDuration - $eventDuration), 1); // 1 as 0 is not valid for the packager
```

Comment:
No sure I understood this - as if (aggregatedDuration > eventDuration) then: $aggregatedDuration - $eventDuration is negative -> entryRoundedDuration - negative is big num (above 1) and I think we should set 1 in this case

Reply:
if "$aggregatedDuration > $eventDuration" is positive (the condition inside if) -> the subtraction "$aggregatedDuration - $eventDuration" is positive, and hence "entryRoundedDuration - POSITIVE" yields number smaller than entryRoundedDuration (or '0' or negative number which will yield '1')

Reply:
I miss understood - then it is good